File: M_Trivia/M_Trivia/GetGameStateResponse.cpp
#include "GetGameStateResponse.hpp"

GetGameStateResponse::GetGameStateResponse(unsigned int status, const GameStatus gameStatus) : Response(status), m_gameStatus(gameStatus)
{
}

nlohmann::json GetGameStateResponse::getJson() const
{
	nlohmann::json j = Response::getJson();
	j["GameStatus"] = this->m_gameStatus;
	return j;
}

File: ClientApp/ClientApp/Models/Responses/GetGameStateResponse.cs
using ClientApp.Models.ResponseErrors;
using Newtonsoft.Json;

namespace ClientApp.Models.Responses
{
    class GetGameStateResponse : Response<NoResponseErrors>
    {
        private GetGameStateResponse() { }

        [JsonProperty]
        public GameStatus GameStatus { get; private set; }
    }
}
